lib/web/join_tokens.go

```diff
@@ -581,6 +592,14 @@ func generateAzureTokenName(rules []*types.ProvisionTokenSpecV2Azure_Rule) (stri
 	return fmt.Sprintf("teleport-ui-azure-%d", h.Sum32()), nil
 }
 
+// generateGitHubTokenName generates a name for the token in the format
+// [bot-name]-[unix-timestamp]. GitHub join tokens will be created via the Machine ID flow.
+// Even if two GitHub join tokens have the same rules, we create different tokens, so
+// users can more easily identify the tokens that enable each bot.
+func generateGitHubTokenName(token types.ProvisionTokenSpecV2) string {
+	return fmt.Sprintf("%s-%d", token.BotName, time.Now().Unix())
```

Comment:
Weren't we going to use the name of the integration rather than the timestamp - or is the BotName already going to be that and the timestamp is just for additional safety?

Reply:
Alright, I've made the following changes:

Instead of using the existing join tokens endpoint, I've added a new endpoint for bots only.
The request will contain the integration name, which will be used to name the join token.

Since the integration name will be propagated not only for the join token, but for the bot and its related roles, using an integration name twice will not be valid, so there's no need to add a timestamp to the resource name.

Please take another look and let me know what you think.